### ISSUE TITLE
Frameworks contains incomplete bitcode

### ISSUE THREAD

**Issue by isacjunior** [flutter/flutter#50205]
## Steps to Reproduce

1. Create a module
2. Create frameworks to embed in iOS application using `flutter build ios-framework --cocoapods`
3. Included Release frameworks and build using [BUCK](https://github.com/facebook/buck).
4. Upload `.ipa` to `TestFlight` using step `upload_to_testflight` in `fastlane` .

 

**Expected results:** 
Success upload to TestFlight

**Actual results:** 
```
[01:16:58]: [Transporter Error Output]: ERROR ITMS-90668: "Invalid Bundle Executable. The executable file 'iOSApp.app/Frameworks/App.framework/App' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu."
[01:16:58]: [Transporter Error Output]: ERROR ITMS-90668: "Invalid Bundle Executable. The executable file 'iOSApp.app/Frameworks/path_provider.framework/path_provider' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu."
[01:16:58]: [Transporter Error Output]: ERROR ITMS-90208: "Invalid Bundle. The bundle iOSApp.app/Frameworks/App.framework does not support the minimum OS Version specified in the Info.plist."
[01:16:58]: Transporter transfer failed.
[01:16:58]: 
[01:16:58]: ERROR ITMS-90668: "Invalid Bundle Executable. The executable file 'iOSApp.app/Frameworks/App.framework/App' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu."
ERROR ITMS-90668: "Invalid Bundle Executable. The executable file 'iOSApp.app/Frameworks/path_provider.framework/path_provider' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu."
ERROR ITMS-90208: "Invalid Bundle. The bundle iOSApp.app/Frameworks/App.framework does not support the minimum OS Version specified in the Info.plist."
```

**flutter version :** v1.13.6

**Comment by jmagman** [comment]
Hopefully fixed with https://github.com/flutter/flutter/pull/49102, which made it into 1.14.4, which is on the beta channel.  Can you run `flutter upgrade` and try again?

**Comment by isacjunior** [comment]
flutter doctor:
```
[✓] Flutter (Channel beta, v1.14.6, on Mac OS X 10.14.6 18G2022, locale en-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.41.1)
[✓] Connected device (1 available)
```
```
flutter build ios-framework --cocoapods
```

```
Building frameworks for com.example.indoor in debug mode...
 ├─Creating Flutter.podspec...                                       2ms

Oops; flutter has exited unexpectedly: "NoSuchMethodError: The method 'getLicenseFile' was called on null.
Receiver: null
Tried calling: getLicenseFile()".
```

**Comment by jmagman** [comment]
I reproduced the `getLicenseFile` crash. `--cocoapods` has a unit test, needs an integration test.

**Comment by jmagman** [comment]
I created https://github.com/flutter/flutter/issues/50364 to track the new crash and write an integration test.

**Comment by thiago-mendes-ifood** [comment]
I solved the problem with this solution, I don't know if it's the best approach, but it works.

**Comment by jmagman** [comment]
> I solved the problem with this solution, I don't know if it's the best approach, but it works.

I don't think that will work, the cache is still nil so [the podspec URL isn't correct.](https://github.com/thim/flutter/commit/2a0599c9c65c4f170145206cae1a370c96f62113#diff-a8643ce1a264062d28b010d237bd51acR267)  I just put out https://github.com/flutter/flutter/pull/50370 for review.  Should be available in master shortly.

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/50370 is now available on the master channel.

**Comment by thim** [comment]
> > I solved the problem with this solution, I don't know if it's the best approach, but it works.
> 
> I don't think that will work, the cache is still nil so [the podspec URL isn't correct.](https://github.com/thim/flutter/commit/2a0599c9c65c4f170145206cae1a370c96f62113#diff-a8643ce1a264062d28b010d237bd51acR267) I just put out #50370 for review. Should be available in master shortly.

My solution solved only the first problem(#50205), I'll merge my code with #50370

**Comment by isacjunior** [comment]
> #50370 is now available on the master channel.

When this solution will be enabled in the `beta` or `stable` channel? The same problem ocorrur using remote `Flutter.framework`.

**Remote Flutter podspec by Google:**
```
Pod::Spec.new do |s|
  s.name                  = 'Flutter'
  s.version               = '1.14.600' # 1.14.6
  s.summary               = 'Flutter Engine Framework'
  s.description           = <<-DESC
Flutter is Google’s UI toolkit for building beautiful, natively compiled applications for mobile, web, and desktop from a single codebase.
This pod vends the iOS Flutter engine framework. It is compatible with application frameworks created with this version of the engine and tools.
The pod version matches Flutter version major.minor.(patch * 100) + hotfix.
DESC
  s.homepage              = 'https://flutter.dev'
  s.license               = { :type => 'MIT', :text => <<-LICENSE
Copyright 2014 The Flutter Authors. All rights reserved.
Redistribution and use in source and binary forms, with or without modification,
are permitted provided that the following conditions are met:
    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above
      copyright notice, this list of conditions and the following
      disclaimer in the documentation and/or other materials provided
      with the distribution.
    * Neither the name of Google Inc. nor the names of its
      contributors may be used to endorse or promote products derived
      from this software without specific prior written permission.
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
LICENSE
  }
  s.author                = { 'Flutter Dev Team' => 'flutter-dev@googlegroups.com' }
  s.source                = { :http => 'https://storage.googleapis.com/flutter_infra/flutter/c4229bfbbae455ad69c967be19aee3fadd6486e1/ios/artifacts.zip' }
  s.documentation_url     = 'https://flutter.dev/docs'
  s.platform              = :ios, '8.0'
  s.vendored_frameworks   = 'Flutter.framework'
  s.prepare_command       = <<-CMD
unzip Flutter.framework -d Flutter.framework
CMD
end
```

**Logs:**
```
ERROR ITMS-90668: "Invalid Bundle Executable. The executable file 'iosApp.app/Frameworks/Flutter.framework/Flutter' contains incomplete bitcode. To compile binaries with complete bitcode, open Xcode and choose Archive in the Product menu."
```

**Comment by jmagman** [comment]
@isacjunior between the fixes in #50370 and #49102 were you able to get your app shipping?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
